core/src/main/java/com/opensymphony/xwork2/ognl/SecurityMemberAccess.java

```diff
@@ -58,14 +58,21 @@ public SecurityMemberAccess(boolean allowStaticMethodAccess) {
         this.allowStaticMethodAccess = allowStaticMethodAccess;
     }
 
-    public boolean getAllowStaticMethodAccess() {
+    public final boolean getAllowStaticMethodAccess() {
```

Comment:
In this particular case it is because `allowStaticMethodAccess` is a private field, so I don't see a valid use case for any descendant class extending this one being allowed to change the behaviour of this getter (and thereby changing the intended behaviour of anything that calls `getAllowStaticMethodAccess()` ).

In more general terms, for other public/protected methods where it seems reasonable to prevent descendants from changing particular behaviour (but who can still call the super method in new methods it introduces), using `final` makes that intent clear ... and Java will attempt to enforce it too.

For variables and attributes, declaring items `final` helps guard against unintentional modification elsewhere (which sometimes helps avoid bugs since the compiler catches it right away).

The above is some of my reasoning anyway ...  :)

Reply:
I think when a method uses a private field doesn't mean it should be `final`. Think about this override for example:
```java
@override
public boolean getAllowStaticMethodAccess() {
    return super.getAllowStaticMethodAccess() || this.getMyOwnLogic();
}
```

Regarding second paragraph, I learnt from @lukaszlenart that it's not a good idea for Struts to decrease public methods flexibility even if they currently don't have public usage in framework because Struts is intended to be extendable and overridable by user.

Regarding third paragraph, yes I feel it's a good idea for local variables :)

Reply:
Thanks for the thoughtful and considered response.  :) :)

True ... having a private field getter doesn't necessarily mean the getter should always be `final`.
In this module, though, the current structure (from constructor through getter) implies that it is a pure "simple" `boolean` flag.  From that point of view it seemed final would make sense (in this specific case).

Your comments about the second paragraph raise a good point as well ... I guess it's more of a consideration as to when extension points should be provided (or limited).  It is certainly something to think about.  :)